src/dynarec/rv64/dynarec_rv64_private.h

```diff
@@ -18,20 +18,28 @@ typedef struct instsize_s instsize_t;
 #define EXT_CACHE_SS     5
 #define EXT_CACHE_SD     6
 #define EXT_CACHE_SCR    7
+#define EXT_CACHE_XMMW   8
+#define EXT_CACHE_XMMR   9
+
 typedef union ext_cache_s {
     int8_t           v;
     struct {
         uint8_t t:4;   // reg type
         uint8_t n:4;   // reg number
     };
 } ext_cache_t;
+
 typedef union sse_cache_s {
-    int8_t      v;
+    int16_t v;
     struct {
-        uint8_t     reg:7;
-        uint8_t     single:1;
+        uint16_t reg : 7;
```

Comment:
7 bits for reg? so a number between 0 and 127? that seems a bit too much.

Reply:
right, there are 16 registers, so 5 bits is enough (an extra bit for not conflict with the special -1), we can keep it in a uint8_t.